cdap-api/src/main/java/co/cask/cdap/api/security/store/SecureStore.java

```diff
@@ -29,15 +29,22 @@
 public interface SecureStore {
 
   /**
+   * List of all the entries in the secure store.
    * @param namespace The namespace that this key belongs to.
    * @return A list of {@link SecureStoreMetadata} objects representing the data stored in the store.
+   * throws NamespaceNotFoundException If the specified namespace does not exist.
```

Comment:
ok I see, all those exceptions are cdap internal.  we are revisiting exception handling across the api and platform in 4.0 so I guess we will address it all then.

DatasetContext has a similar issue, it just throws DatasetInstantiationException everywhere. PluginContext throws an IllegalArgumentException if the plugin id doesn't exist.